### PR TITLE
JSAB2-74 Navigation

### PR DESCRIPTION
**Pull request by hongtaoTang**
Create bottom tab navigation:
My kingdom tab --> /kingdom/buildings (mocked)
Settings -->  /settings page (mocked)

### CODE REVIEW

File: frontend/src/Buildings/index.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { View, Text } from 'react-native';
+
+function Buildings() {
+  return (
+    <>
+      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+        <Text>Mocked Buildings</Text>
+      </View>
+    </>
```
Comment (Kyya) [inline frontend/src/Buildings/index.jsx]: Is it necessary to have `<>` and `</>` here?

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
+const AppContainer = createAppContainer(TabNavigator);
 
 function App() {
   return (
     <Provider store={store}>
-      <View style={commonStyles.container}>
-        <Menu />
-        <Troop />
-      </View>
+      <AppContainer />
```
Comment (evasimonyi) [inline frontend/src/App.jsx]: Screen Shot 2019-12-11 at 11 56 40

Active tab button looks like this. Please research if you can support safeArea, check if it's possible to highlight the active nav button in the safeArea as well.

File: frontend/src/Navigation/index.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import Ionicons from 'react-native-vector-icons/Ionicons';
+import Settings from '../Settings';
+import Buildings from '../Buildings';
+import Colors from '../common/colors';
+
+const TabNavigator = createBottomTabNavigator({
+  'My Kingdom': {
+    screen: Buildings,
+    navigationOptions: {
+      tabBarIcon: () => <Ionicons name="ios-home" size={25} style={{ color: 'white' }} />,
```
Comment (evasimonyi) [inline frontend/src/Navigation/index.jsx]: please add white to the `common/colors.js` and import it from there, we'll use it everywhere

File: frontend/src/Navigation/index.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  initialRouteName: 'My Kingdom',
+  order: ['My Kingdom', 'Settings'],
+  tabBarOptions: {
+    activeBackgroundColor: '#00695c',
+    tabStyle: {
+      padding: 0,
+    },
+    labelStyle: {
+      fontWeight: 'bold',
+      color: 'white',
```
Comment (evasimonyi) [inline frontend/src/Navigation/index.jsx]: please import this color from the `common/colors.js` so it'll be easier to modify it in the future

File: frontend/src/App.jsx (hunk 2 above)
Comment (hongtaoTang) [reply frontend/src/App.jsx]: It seems like I cant apply any styles to the safeArea, but I can disable the bottom-safeArea for all kinds of devices. Should i do that?

Comment (evasimonyi) [reply frontend/src/App.jsx]: no, don't do that

Comment (evasimonyi) [reply frontend/src/App.jsx]: it's OK this way then.

**Comment by Kyya** [comment]
I got this warning when try to running...
```
Some of your project's dependencies are not compatible with currently installed expo package version:
 - **expo-constants** - expected version range: ~7.0.0 - actual version installed: ^8.0.0
Your project may not work correctly until you install the correct versions of the packages.
To install the correct versions of these packages, please run: expo install [package-name ...]
```

File: frontend/package.json [hunk 5]
```diff
@@ -10,13 +10,19 @@
   },
   "dependencies": {
     "expo": "^35.0.0",
+    "expo-constants": "^8.0.0",
```
Comment (Kyya) [inline frontend/package.json]: maybe it's too new for the expo sdk.
I got this
```
Some of your project's dependencies are not compatible with currently installed expo package version:
 - expo-constants - expected version range: ~7.0.0 - actual version installed: ^8.0.0
Your project may not work correctly until you install the correct versions of the packages. To install the correct versions of these packages, please run: expo install [package-name ...]
```

Comment (hongtaoTang) [reply frontend/package.json]: thx Kyya

File: frontend/package.json [hunk 6]
```diff
@@ -10,13 +10,19 @@
...
   "dependencies": {
     "expo": "^35.0.0",
+    "expo-constants": "~7.0.0",
     "native-base": "^2.13.8",
     "prop-types": "^15.7.2",
     "react": "16.8.3",
     "react-dom": "16.8.3",
     "react-native": "https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz",
+    "react-native-gesture-handler": "~1.3.0",
+    "react-native-reanimated": "~1.2.0",
```
Comment (evasimonyi) [inline frontend/package.json]: where do you use it? what is this for?

File: frontend/src/Navigation/myKingdom.jsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (evasimonyi) [inline frontend/src/Navigation/myKingdom.jsx]: great idea for creating a basic 'layout', we can use this later!
but please don't put it into the navigation folder, you didn't put the Settings page here either!
create a folder for it, like MainPage, or something like that.

File: frontend/src/Navigation/index.jsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+import Ionicons from 'react-native-vector-icons/Ionicons';
+import Settings from '../Settings';
+import Colors from '../common/colors';
+import MyKingdom from './myKingdom';
+
+const TabNavigator = createBottomTabNavigator({
+  MyKingdom: {
+    screen: MyKingdom,
+    navigationOptions: {
+      tabBarIcon: () => <Ionicons name="ios-home" size={25} style={{ color: Colors.whiteColor }} />,
```
Comment (evasimonyi) [inline frontend/src/Navigation/index.jsx]: too long line, please break it

File: frontend/src/Navigation/index.jsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+      tabBarLabel: 'My Kingdom',
+    },
+  },
+
+  Settings: {
+    screen: Settings,
+    navigationOptions: {
+      tabBarIcon: () => <Ionicons name="md-settings" size={25} style={{ color: Colors.whiteColor }} />,
```
Comment (evasimonyi) [inline frontend/src/Navigation/index.jsx]: too long line, please break it

File: frontend/src/Settings/index.jsx [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { View, Text } from 'react-native';
+import Styles from '../common/styles';
+
+function Settings() {
+  return (
+    <View style={[Styles.screenStyle, { flex: 1, justifyContent: 'center', alignItems: 'center' }]}>
```
Comment (evasimonyi) [inline frontend/src/Settings/index.jsx]: too long line, please break it

File: frontend/package.json (hunk 6 above)
Comment (evasimonyi) [reply frontend/package.json]: please don't delete immediately, I'm just not sure what is this for, I don't see this being used anywhere

**Pull request merged**